src/java.base/share/classes/sun/security/rsa/RSACore.java

```diff
@@ -145,7 +163,9 @@ public static int getByteLength(RSAKey key) {
     public static byte[] rsa(byte[] msg, RSAPrivateKey key, boolean verify)
             throws BadPaddingException {
         if (key instanceof RSAPrivateCrtKey) {
-            if (useNativeRsa && key instanceof sun.security.rsa.RSAPrivateCrtKeyImpl) {
+            if (useNativeRsa
+                && key instanceof sun.security.rsa.RSAPrivateCrtKeyImpl
+                && ((NativeCrypto.getVersion() != 0) || (checkKeySizeSupport(key.getModulus().bitLength())))) {
```

Comment:
Redundant parentheses in `(checkKeySizeSupport(key.getModulus().bitLength()))`.

Reply:
Okay, I removed them.